### PR TITLE
iio_wifi_backend 4: Iio network support in iio and iio_demo

### PR DESCRIPTION
**Pull request by mchindri**
- Add iio network support using tcp_socket
- Add network support on iio_demo

**Last 2 commits** are relevant
Will be rebased when network commits are merged

### CODE REVIEW

**Comment by mchindri** [comment]
> With the latest changes on master branch, new conflicts appeared. Please fix them

will rebase after #698 is merge

File: libraries/iio/iio.c [hunk 1]
```diff
@@ -98,11 +113,107 @@ static struct iio_desc		*g_desc;
...
+
+	/* Get all new waiting sockets.
+	 * If none is available, wait until first connection */
+	do {
+		ret = socket_accept(desc->server, &sock);
+		if (ret == -EAGAIN) {
+			nb_active_sockets = _nb_active_sockets(desc);
+			if (nb_active_sockets == 0)
+				/* Wait until a connection exists */
+				continue;
```
Comment (buha) [inline libraries/iio/iio.c:153]: I would add a debouncing delay right before continue, something like `mdelay(1)`.

Comment (dbogdan) [reply libraries/iio/iio.c:153]: I think is a good idea too.

**Comment by mchindri** [comment]
v2:
 - Rebase onto master.
 - Add mdelay(1)

File: libraries/iio/iio.c (hunk 1 above)
Comment (mchindri) [reply libraries/iio/iio.c:183]: done

**Pull request merged**